### PR TITLE
Updated sprite names to conform to PokeAPI's id numbering schemes for…

### PR DESCRIPTION
**Pull request by zacbullard**
… non-default pokemon forms. Also added some missing sprites, courtesy of Smogon.

A comprehensive before and after file name comparison can be found in the following .csv:
[sprite_before_after.txt](https://github.com/PokeAPI/sprites/files/5864798/sprite_before_after.txt)

Miscellaneous details:
 
There were several "totem" pokemon icons that I was unsure where their proper home should be, so I didn't touch them. For instance, araquanid-totem is not in sprites/sprites/pokemon, but it is in sprites/sprites/pokemon/versions/generation-vii/icons.

Zacian and Zamazenta's default sprites were erroneously of their alternate forms ("Crowned"), so I shifted over 888.png and 889.png to 10181.png and 10182.png and added in their true default form ("Hero of Many Battles") sprites.

All added sprites were taken from the indicated Smogon Google Sheet: https://docs.google.com/spreadsheets/d/1acgzAjh0dnFRQnjZu8kSjS177rKCzpFfEHRLtwuuXRU/edit#gid=0

### CODE REVIEW

**Comment by Naramsim** [comment]
Great work!

I'll also review it. I'll ask also @C-Garza

**Review by Naramsim — APPROVED** [review]

**Comment by C-Garza** [comment]
Renaming the images can cause issues with the `pokemon-form` endpoint, which is why I suggested copying the files and renaming those copies. The sprites for pokemon forms are built differently, as you can see in this endpoint for [`raichu-alola`](https://pokeapi.co/api/v2/pokemon-form/10202).

When I was adding sprites I wondered why there was this duplication of sprite images if forms have their own unique pokemon id number. 

Here is a link to the function that makes the pokemon form sprites in [the build file](https://github.com/PokeAPI/pokeapi/blob/1ce88ff23d00831bd160bb3eef2614c6f64b9f0d/data/v2/build.py#L2033-L2058). If every form has a unique pokemon id then I think we could just grab that id instead and safely rename the files in this pull request. I think all that if else logic could be replaced with a single line: 
`file_name = "%s.png" % getattr(pokemon, "id")`.

I'm just not sure if every pokemon form has unique pokemon id's. What do you think @Naramsim?

**Comment by Naramsim** [comment]
I see @C-Garza. I think every pokemon form is associated with a single Pokemon.

Then maybe it's better to change the logic of the API. Do you think also the duplication would be required?

**Comment by C-Garza** [comment]
So I went ahead and checked the [pokemon_forms.csv](https://github.com/PokeAPI/pokeapi/blob/master/data/v2/csv/pokemon_forms.csv) to see if there were any forms that didn't have unique `pokemon_id`'s and there were quite a few. (examples: [unown](https://github.com/PokeAPI/pokeapi/blob/master/data/v2/csv/pokemon_forms.csv#L900) and [arceus](https://github.com/PokeAPI/pokeapi/blob/master/data/v2/csv/pokemon_forms.csv#L940))

That makes changing the logic for the API a bit harder. Some forms that do have unique `pokemon_id`'s are `mega`, `alola`, `galar`, `gmax`, and `totem`. The [build file](https://github.com/PokeAPI/pokeapi/blob/1ce88ff23d00831bd160bb3eef2614c6f64b9f0d/data/v2/build.py#L2036) already uses a regex to target `mega` forms and use their `pokemon_id`, we could expand it to include other forms that have a unique `pokemon_id`, but it feels weird to target specific forms.

I can't really think of a good solution for removing the duplicates if all the forms don't have unique `pokemon_id`'s, unless we give all the pokemon forms without a unique id one in the .csv files. 

If the logic can't be changed, then I don't think these images should be renamed but copied and renamed, or else the `pokemon-form/` endpoint loses their sprites.

**Comment by zacbullard** [comment]
Alright then, I can close this pull request and put in a new pull request, using copied files and not renamed ones.

I thought that having a bunch of duplicated files would be wasteful and potentially confusing, but it seems that there are underlying architectural reasons that aren't easy to update.

**Comment by C-Garza** [comment]
@zacbullard Wait a second, I just noticed the column `is_default` in the [pokemon_forms.csv](https://github.com/PokeAPI/pokeapi/blob/master/data/v2/csv/pokemon_forms.csv) file is `0` when the pokemon form does not have a unique `pokemon_id`. I think we can rename the files now if we work with this logic in the `build.py` file.

**Comment by zacbullard** [comment]
Ok @C-Garza ! I'll just keep an eye on the discussion here then.

**Comment by C-Garza** [comment]
Alright, I think I got it. In the [build.py](https://github.com/PokeAPI/pokeapi/blob/1ce88ff23d00831bd160bb3eef2614c6f64b9f0d/data/v2/build.py#L2033-L2058) file we can replace the if-else logic with something like this:
```py
if info[5] == "1": # If this form is the default form, return pokemon_id
    file_name = "%s.png" % getattr(pokemon, "id")
else: # If this form is not the default, return id with form name (ex. unown-b)
    file_name = "%s-%s.png" % (
        getattr(pokemon, "pokemon_species_id"),
        info[2],
    )
```
I think with this logic we can remove duplicates @Naramsim. I tested it out locally and it seems to work just fine. If they have a unique pokemon id, then that is used, if they don't, then it just uses the same behavior. 

Everything in this pull request has a unique id, so I think it would be safe to rename them and accept this pull request if we make this change in the build.py file.

**Comment by Naramsim** [comment]
@C-Garza, right now the situation is a mess if we talk about variants/forms/etc. In the repository, we have files like `10014.png` and `351-rainy.png` for the same Castform rain.

Now I think we shouldn't have any duplication of files. Otherwise it will become a worse mess. I came up with a solution that builds the sprite's URL from the name OR the ID. For both the `/pokemon` and `/pokemon-form` endpoints.

Basically it takes whichever image is available.

Can you check it out locally on your machine @C-Garza? Keep in mind that it's super dirty code and there's duplication of code there. It's just a starting point. https://github.com/PokeAPI/pokeapi/pull/568/files

PS: for my tests I used `pikachu-hoenn-cap`.

**Comment by C-Garza** [comment]
@Naramsim, what I'm suggesting is that we take all those forms with a unique pokemon id and rename them, like `351-rain.png` and `pikachu-hoenn-cap` to `10014.png` and `10095.png`. That will remove the duplicates and the `pokemon` endpoint would be able to get pokemon form sprites. [`pikachu-hoenn-cap`](https://pokeapi.co/api/v2/pokemon/pikachu-hoenn-cap) looks for `10014.png` which is why it returns null for it's call currently.

The only thing we would have to change at that point is the logic in getting the sprites for the `pokemon-form` endpoint in the [build.py](https://github.com/PokeAPI/pokeapi/blob/1ce88ff23d00831bd160bb3eef2614c6f64b9f0d/data/v2/build.py#L2035-L2044) file, which I think I have done with the code here:
```py
if info[5] == "1": # If this form is the default form, return pokemon_id
    file_name = "%s.png" % getattr(pokemon, "id")
else: # If this form is not the default, return id with form name (ex. unown-b)
    file_name = "%s-%s.png" % (
        getattr(pokemon, "pokemon_species_id"),
        info[2],
    )
```

I tested this locally and it seems to resolve the issue I believe. 

I think this solution will be easier in the long run because sprites will now be named after their unique pokemon id unless a form doesn't have one, in that case we append the species id and the form name, like it was doing before, an example of one without a unique id being `unown-b` which would have a sprite of `201-b.png`.

**Comment by Naramsim** [comment]
If we are able to rename all the sprites correctly then your solution will be the cleanest.

The only possible downside I see is when Veekun will finally release updated data and they'll likely follow the `10xxx.png` naming convention. But I'm ok with it.

So if we have a volunteer that will rename all `10xxx.png` sprites to `ID-variant.png` we can use your clean solution.

**Comment by C-Garza** [comment]
Updated data for gen 8 forms you mean? If Veekun fills out the appropriate columns in the `pokemon_forms.csv` then I think this would still work. We could always still implement the check you wrote as a safeguard just in case.

**Comment by Naramsim** [comment]
Initially, I thought you wanted to rename the sprites `10xxx.png` to `ID-variant.png`. But I think you want to do the opposite. If I understand it well you are suggesting to rename part of the `ID-variant.png` to `10xxx.png`, but only for those forms which are defaults. If one is not a default form, then append an identifier.

If it's so, I don't see a clear simplification. 

---

This is what I would do to have a naming system that is clean:

I'd rename all `10xxx.png` sprites to `ID-variant.png`. So for example, each Castform form will have only one corresponding `351-<weather>.png`. Pikachu-hoenn-cap will have only `pikachu-hoenn-cap.png`. We will delete as well all the sprites `10xxx.png`.

The only possible downside I see is when Veekun will finally release updated data and they'll likely follow the `10xxx.png` naming convention. But I'm ok with it.

So if we have a volunteer that will rename all `10xxx.png` sprites to `ID-variant.png` this is the cleanest solution. Then we would also need to modify a bit the code of the API.

---

Still, I like my solution because it allows both `10xxx.png` and `ID-variant.png` to coexist in this repository. Without having to go through a having renaming. (and without having developers change the links of their images, I assume many of them have hardcoded values)

**Comment by C-Garza** [comment]
Yeah I wanted to change `ID-variant.png` to `10xxx.png` if they have a `10xxx` id, and leave them as `ID-variant.png` if they don't have a `10xxx` id. The reason being is because the `pokemon` endpoint looks for the pokemon id, which includes the `10xxx`, and if most of the forms have a pokemon id like `10xxx`, then we can do the same thing for the `pokemon-form` endpoint. And if they don't, then we use the `ID-variant.png`.

We could do it your way. I was thinking about what would happen to the people who have it hardcoded as well and your solution wouldn't ruin that.

**Comment by zacbullard** [comment]
So it sounds like you guys are going for mixed naming conventions for default forms, with a preference for not renaming files. From my experience I feel like no one is actually hard-coding the ```ID-variant.png``` format of the names specifically for the files that I've renamed here, as they currently aren't even being expressed in the API (this is what drove me to rename these files in the first place, so I could actually call them!). Unless someone out there is bypassing the API entirely and hardcoding links directly to the ```https://raw.githubusercontent.com/...``` sprite urls, which seems like a bad idea and defeats the purpose of having a RESTful API in the first place?!

Either way, I can convert the sprite names to any desired format. You're the boss. If the two of you are going for the mixed convention and really don't want the renaming, I can cancel this pull request here and submit another with simply the missing images added, if you two want.

A possible issue I've seen is that it seems that the "model" files for variants are not in their proper folder. For instance, currently Mega Venusaur's model file, ```3-mega.png```, is in ```sprites/sprites/pokemon```, and in this same directory the _sprite_ file for Mega Venusaur also exists, ```10033.png```. This might be an issue for the mixed format parsing code, as in the folder right now there are mixed case files with different rendering styles present. It seems the proper home for the models is ```sprites/sprites/pokemon/model```, but no non-default variants live there at the moment.

**Comment by Naramsim** [comment]
Hi everyone, sorry for the late reply but I don't have time these days. 

@zacbullard, I see your points. But still, I wouldn't like to rename/move images. I'd like to continue with my option and leave this PR on hold. So maybe you could really open another PR just to add the missing images you were talking about and also for correcting Zacian and Zamazenta.

Also, the `model` folder is not used and it was never used. They look like the sprites released in gen6 and gen7 but with a better image quality. What do you think?

**Comment by zacbullard** [comment]
Hey, sorry for the late reply, I got busy as well. I spent a bit more time thinking about the current situation.

The "model" images are the 3D rendered models of the pokemon [https://github.com/PokeAPI/sprites/blob/master/sprites/pokemon/3-mega.png](like the this Mega Venusaur I mentioned, `3-mega.png`) while the rest of the images are 2D sprites [https://github.com/PokeAPI/sprites/blob/master/sprites/pokemon/10033.png](like the this other Mega Venusaur I mentioned,`10033.png`). Interesting the model folder is never used, maybe at some point in the past there was an intention to add a 3D image option for the API in addition to the current 2D ones?

Do you see my point about how for Mega Venusaur there is a collision in which image to use? That is, both `3-mega.png' and `10033.png` exist in the same folder, but are different images, so  depending on how your mixed format parser works, one file will be given preference over the other. And this issue exists for many other forms as well.

The current image situation is a mess with the mixed conventions, collisions, and never-used files. In my opinion, normalizing the convention by accepting this pull request (which uses the Veekun standard format) is the cleanest solution. @Naramsim, can you elaborate on your fears of renaming the files? If this majorly disrupts a contingent of users who have been bypassing the api and hardcoding the direct github urls (which is a very unusual and bad convention to follow!), then we can always revert this commit. The benefits of accepting this pull far outweigh the cons.

I can also help sort out collisions and the modeling files, but this pull would have to be accepted first before addressing those other issues.

**Comment by Naramsim** [comment]
Hi Zack, so I came up with my final solution that supports both naming conventions. I also noted that in some folders for the same pokemon/form we have both the `10xxx.png` and `ID-form.png` files. I also noted that generally the `10xxx.png` looks better and has the art correct. For example, take Mewtwo, [10043.png](https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10043.png) is the better image while [150-mega-x.png](https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150-mega-x.png) is out of focus and the art is wrong. 

So with my solution, I decided to give priority to `10xxx.png` files which goes towards what you are also proposing.

Ah, another thing that popped up when I was testing Mewtwo is that we have don't have the best-looking sprite for the shiny variant. In fact, we only have the [shiny/150-mega-x.png](https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150-mega-x.png). I suspect this is happening for many other pokemon.

So, first I want to have my PR merged over at PokeAPI, and I'd like to ask you @zacbullard and @C-Garza if you could review it a bit. Maybe just by looking at the API deployed in staging.

https://staging.pokeapi.co/api/v2/pokemon/10043/
https://staging.pokeapi.co/api/v2/pokemon-form/10143/

Then @zacbullard, we can go with your renaming and bring as many `ID-form.png`s to the Veekun format.

**Comment by zacbullard** [comment]
Great! Sounds like a plan. I'll take a look...

**Comment by Naramsim** [comment]
Ok, let's start with this PR. I don't know where you wanna start but I want to say a thing. 

If we are adding or changing the names of our main pokemon sprites we should use the one of Smogon, they were very kind and allowed us to serve them. In part for the standard front pokemon, we are already doing it.

- gen-8 https://www.smogon.com/forums/threads/sword-shield-sprite-project.3647722/
- gen 7 https://www.smogon.com/forums/threads/sun-moon-sprite-project.3577711/
- gen 6 https://www.smogon.com/forums/threads/x-y-sprite-project.3486712/

They do provide female, shiny, back variants. But the naming format is pretty different from ours. I don't know if you wanna start from those, but I think those are the most important since are the ones who live in our root folder. Here we have two options: renaming them to one of our naming conventions or code again the API for checking a different name.

Ah, if you are not familiar with it @C-Garza @zacbullard, the default pokemon sprites we serve should be the gen-5 version. The Smongon guys have a team of more than 20 people who created for gen-6, gen-7, and gen-8 their gen-5 sprite art.

**Comment by zacbullard** [comment]
Yeah, I noticed there is a lot of potential to adding shiny/back sprites to the API. It'll take a little effort to convert the sprite names from the Smogon style to the Veekun style, but I think that's probably the most internally consistent thing to do for PokeAPI to match everything else. 

Filling out the front defaults in this PR is a good place to start. Well, the staging still looks good to me, everything should be set, anything else you want to look at in your review for this PR @C-Garza ?

**Review by C-Garza — APPROVED** [review]
The changes here to `10xxx.png` look to be correct to me.

**Comment by C-Garza** [comment]
I think with some scripts we can change the names of most of the smogon images. The ones with forms will need a bit more work though.
